examples/neural_compressor/optical-character-recognition/run_trocr.py

```diff
@@ -0,0 +1,310 @@
+            pred_str = processor.batch_decode(outputs, skip_special_tokens=True)
+            labels = batch["labels"]
+            labels[labels == -100] = processor.tokenizer.pad_token_id
+            label_str = processor.batch_decode(labels, skip_special_tokens=True)
+
+            # add batch to metric
+            cer.add_batch(predictions=pred_str, references=label_str)
+            if iters is not None and i >= iters:
+                break
+        score = cer.compute()
+        print("Batch size = %d" % training_args.per_device_eval_batch_size)
+        if training_args.per_device_eval_batch_size == 1:
+            print("Latency: %.3f ms" % (batch_time.avg * 1000))
+        print("Throughput: %.3f images/sec" % (training_args.per_device_eval_batch_size / batch_time.avg))
+
+        # TODO: this should also be done with the ProgressMeter
+        print("cer: {score:.5f}".format(score=(score)))
+
+        return score
+
+    if optim_args.apply_quantization:
+        torch.backends.quantized.engine = "onednn"
+        default_config = os.path.join(
+            os.path.abspath(os.path.join(__file__, os.path.pardir, os.path.pardir)), "config"
+        )
+        q8_config = IncQuantizationConfig.from_pretrained(
+            optim_args.quantization_config if optim_args.quantization_config is not None else default_config,
+            config_file_name="quantization.yml",
+            cache_dir=model_args.cache_dir,
+        )
+        # Set metric tolerance if specified
+        if optim_args.tolerance_criterion is not None:
+            q8_config.set_tolerance(optim_args.tolerance_criterion)
+        # Set quantization approach if specified
+        if optim_args.quantization_approach is not None:
+            supported_approach = {"static", "dynamic", "aware_training"}
+            if optim_args.quantization_approach not in supported_approach:
+                raise ValueError(
+                    "Unknown quantization approach. Supported approach are " + ", ".join(supported_approach)
+                )
+            quant_approach = getattr(IncQuantizationMode, optim_args.quantization_approach.upper()).value
+            q8_config.set_config("quantization.approach", quant_approach)
+        quant_approach = IncQuantizationMode(q8_config.get_config("quantization.approach"))
+        # torch FX used for post-training quantization and quantization aware training
+        # dynamic quantization will be added when torch FX is more mature
+        if quant_approach != IncQuantizationMode.DYNAMIC:
+            q8_config.set_config("model.framework", "pytorch_fx")
+        q8_config.set_config("tuning.accuracy_criterion.higher_is_better", False)
+        quantizer = IncQuantizer(q8_config, eval_func=eval_func, calib_dataloader=test_dataloader)
+        optimizer = IncOptimizer(model, quantizer=quantizer)
+        q_model = optimizer.fit()
+        optimizer.save_pretrained(training_args.output_dir)
+
+    if optim_args.verify_loading:
```

Comment:
To verify the quantized model loading, I would add an evaluation step in order to compare it with the one of `q_model` (that what we currently do in the other examples scripts) which could also be replaced with inference. If not we should remove the `verify_loading` flag in my opinion.